raiden/api/rest.py

```diff
@@ -257,6 +258,16 @@ def open(self, partner_address, token_address, settle_timeout, balance=None):
             )
         except (InvalidAddress, InvalidSettleTimeout, SamePeerAddress) as e:
             return make_response(str(e), httplib.CONFLICT)
+        except ValueError as e:
+            if re.match("Token address 0x[0-9a-f]{40} does not contain code", str(e)):
+                return make_response(str(e), httplib.CONFLICT)
+            elif re.match("Manager for token 0x[0-9a-f]{40} does not exist", str(e)):
+                return make_response(str(e), httplib.CONFLICT)
+            raise
+        except Exception as e:
+            if re.match("^Duplicated channel$", str(e)):
```

Comment:
Right, didn't remember it had been deprecated in Py3. Anyway, I'd suggest not trusting in `str` conversions, as it may be reimplemented in inherited classes. Maybe, `e.args[0]` would be closer to the pattern.